pandas/util/testing.py

```diff
@@ -536,23 +537,128 @@ def assert_equal(a, b, msg=""):
     assert a == b, "%s: %r != %r" % (msg.format(a,b), a, b)
 
 
-def assert_index_equal(left, right, exact=False, check_names=True):
+def assert_index_equal(left, right, exact=False, check_names=True,
+                       check_less_precise=False, check_exact=True, obj='Index'):
+    """Check that left and right Index are equal.
+
+    Parameters
+    ----------
+    left : Index
+    right : Index
+    exact : bool, default False
+        Whether to check the Index class, dtype and inferred_type are identical.
+    check_names : bool, default True
+        Whether to check the names attribute.
+    check_less_precise : bool, default False
+        Specify comparison precision. Only used when check_exact is False.
+        5 digits (False) or 3 digits (True) after decimal points are compared.
+    check_exact : bool, default True
+        Whether to compare number exactly.
```

Comment:
I'm willing to change if there is better wording / options. Do you have any idea? There are 3 levels:

- 1: Use assert_equal
- 2: Use assert_almost_equal, comparing 5 digits after decimal points (default)
- 3: Use assert_almost_equal, comparing 3 digits after decimal points